cli/keyring.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"crypto/rand"
+	"errors"
+	"syscall"
+
+	"github.com/99designs/keyring"
+	"github.com/libp2p/go-libp2p/core/crypto"
+	"github.com/spf13/cobra"
+	"golang.org/x/term"
+)
+
+const (
+	peerKeyName             = "peer_key"
+	badgerEncryptionKeyName = "badger_encryption_key"
+)
+
+// openKeyring attempts to open the keyring file from the given directory.
+//
+// If the directory is an empty string the best option for the current OS will be used.
+func openKeyring(cmd *cobra.Command, dir string) (keyring.Keyring, error) {
+	var allowedBackends []keyring.BackendType
+	if dir != "" {
+		// only allow file backend if a directory is specified
+		allowedBackends = append(allowedBackends, keyring.FileBackend)
+	}
+
+	prompt := keyring.PromptFunc(func(s string) (string, error) {
+		cmd.Print(s)
+		pass, err := term.ReadPassword(int(syscall.Stdin))
+		return string(pass), err
+	})
+
+	return keyring.Open(keyring.Config{
+		AllowedBackends:         allowedBackends,
+		ServiceName:             "defradb",
+		KeychainName:            "defradb",
+		KeychainPasswordFunc:    prompt,
+		FilePasswordFunc:        prompt,
+		FileDir:                 dir,
+		KeyCtlScope:             "user",
+		KeyCtlPerm:              0, // TODO
```

Comment:
todo: Please create a task if this is something that needs to be handled later. Otherwise it will linger.

Reply:
I will fix before merging.

Reply:
I ended up replacing this dependency.